### PR TITLE
Fix dialyzer warnings 

### LINKED ISSUE #71: Type specs issue with Dialyzer
Code:
```
case :eredis_cluster.qa :db, ["PING"] do
  [ok: "PONG"] -> true
  _ -> false
end
```

Error:
```
The pattern can never match the type.

Pattern:
[{:ok, <<80, 79, 78, 71>>}]

Type:

  [
    {:error, :invalid_cluster_command | :no_connection | :tcp_closed | bitstring()}
    | {:ok, :undefined | [:undefined | bitstring()]}
  ]
  | {:error, :no_connection}
```

Code works correctly, however `"PONG"` doesn't match `:undefined | [:undefined | bitstring()]` typespecs.

Erlang/OTP 25
Elixir (1.14.2)

### PR DESCRIPTION
**Pull request by bjosv**
Run dialyzer on the test profile during make dialyzer, and in CI runs.
This PR fixes all dialyzer warnings found in the rebar3 test profile.

Fixes #71

### CODE REVIEW

File: include/eredis_cluster.hrl [hunk 1]
```diff
@@ -6,10 +6,11 @@
 
 -type redis_error_result() :: Reason::bitstring() | no_connection
     | invalid_cluster_command | tcp_closed.
--type redis_success_result() :: Result::bitstring() | undefined.
+-type redis_success_result() :: bitstring() | [bitstring()] | undefined.
```
Comment (zuiderkwast) [inline include/eredis_cluster.hrl:9]: Where `bitstring()` (any multiple of bits) is used in these types, it should really be `binary()` (a multiple of 8 bits).

**Review by zuiderkwast — COMMENTED** [review]
I suppose silencing the warnings is better than nothing, but these types are a bit of a mess.

File: include/eredis_cluster.hrl (hunk 1 above)
Comment (bjosv) [reply include/eredis_cluster.hrl:9]: Sounds reasonable, but I got stuck when looking into why the change was done in https://github.com/Nordix/eredis_cluster/commit/071b73be75561b370b807f909c8a8899618c951d.
Maybe it was in early development stages and was then just kept? Can an `EVAL` return anything else in Redis?

Looking at eredis v1.0.8, used at the time of the change, it states `binary()` in the `return_value()` and no `bitstring()` mentioned in the eredis code either. Sure looks like we should change this.

Comment (zuiderkwast) [reply include/eredis_cluster.hrl:9]: EVAL can return RESP data, like any other redis command. It's not even possible to receive parts of a byte over TCP.

Possibly some other spec used bitstring() which gave some warning when it was used with binary(). Do you know if there were any occurrences of bitstring prior to that change?

Comment (bjosv) [reply include/eredis_cluster.hrl:9]: No bitstrings can be found in the repo prior to that change.

**Review by zuiderkwast — APPROVED** [review]

**Pull request merged**